app.go

```diff
@@ -438,3 +449,89 @@ func executeCancelSell(cmd command) bool {
 
 	return true
 }
+
+func executeSetBuyAmount(cmd command) bool {
+	userID := cmd.UserID
+	strAmount = cmd.Args[1]
+	stock := cmd.Args[0]
+	account := accountStore.GetAccount(userID)
+
+	if account == nil {
+		log.Infof("User %s does not have an account", userID)
+		return false
+	}
+
+	amount, err := currency.NewFromString(strAmount)
+	if err != nil {
+		log.Error("Failed to parse currency")
+		return false
+	}
+	err = account.RemoveFunds(amount)
+	if err != nil {
+		log.Error("User had insufficient funds to set buy amount of %s", amount)
+		return false
+	}
+	autoBuyRequestStore.AddAutorequest(cmd.Args[0], cmd.UserID, amount)
+	log.Infof("User %s set automated buy amount for %s dollars of stock %s", userID, amount, stock)
+	return true
+}
+
+func executeSetSellAmount(cmd command) bool {
+	userID := cmd.UserID
+	strAmount = cmd.Args[1]
+	stock := cmd.Args[0]
+	account := accountStore.GetAccount(userID)
+
+	if account == nil {
+		log.Infof("User %s does not have an account", userID)
+		return false
+	}
+
+	amount, err := currency.NewFromString(strAmount)
+	if err != nil {
+		log.Error("Failed to parse currency")
+		return false
+	}
+	autoSellRequestStore.AddAutorequest(stock, userID, amount)
+	log.Infof("User %s set automated sell amount for %s dollars of stock %s", userID, amount, stock)
+	return true
+}
+
+func executeCancelSetBuy(cmd command) bool {
+	userID := cmd.UserID
+	stock := cmd.Args[0]
+	account := accountStore.GetAccount(userID)
+
+	if account == nil {
+		log.Infof("User %s does not have an account", userID)
+		return false
+	}
+
+	refund := autoBuyRequestStore.CancelAutorequest(stock, userID)
+	if refund == -1.0{
```

Comment:
We should return an error object instead of checking for an arbitrary return value.